### ISSUE TITLE
Delete user fails with ConcurrencyFailure

### ISSUE THREAD

**Issue by marcuslindblom** [JudahGabriel/RavenDB.Identity#8]
I use the `6.0.1` and when I try to delete a user I get the [error](https://github.com/JudahGabriel/RavenDB.Identity/blob/master/RavenDB.Identity/UserStore.cs#L202) `Unable to delete user email compare/exchange value`.  I use RavenDB `4.1.3`.

**Comment by marcuslindblom** [comment]
According to the documentation it seems like you first need to get the current value and then use the index when you delete the key.

From the docs
```c#
// First, get existing value
CompareExchangeValue<User> readResult =
    store.Operations.Send(
        new GetCompareExchangeValueOperation<User>("AdminUser"));

// Delete the key - use the index received from the 'Get' operation
CompareExchangeResult<User> deleteResult
    = store.Operations.Send(
        new DeleteCompareExchangeValueOperation<User>("AdminUser", readResult.Index));

// The delete result is successful only if the index has not changed between the read and delete operations
bool deleteResultSuccessful = deleteResult.Successful;
```
https://ravendb.net/docs/article-page/4.1/csharp/client-api/operations/compare-exchange/delete-compare-exchange-value

**Comment by JudahGabriel** [comment]
Marcus, I just tested this using the sample project, and it worked.

Can you confirm:

- Was the user created with a previous version of RavenDB.Identity? If so, [follow the migration steps](https://github.com/JudahGabriel/RavenDB.Identity/#updating-from-old-version).
- If this user was created using the latest (6.x) version of RavenDB.Idenity, is there an existing compare/exchange value in the database? You should see something like: emails/foo@bar.com, where foo@bar.com is the email address of the user you're trying to delete.

**Comment by JudahGabriel** [comment]
OK Marcus, this is fixed. I've published an update to NuGet (v6.0.2) as well.

Thanks for the bug report.
